### PR TITLE
refactor(api): RPC > HTTP /save/

### PR DESCRIPTION
**Pull request by Arqu**

### CODE REVIEW

File: lib/api.go [hunk 1]
```diff
@@ -133,3 +134,22 @@ const (
 	// AEWebUI serves the remote WebUI
 	AEWebUI = APIEndpoint("/webui")
 )
+
+// DatasetRefFromPath parses a path and returns a datasetRef
+func DatasetRefFromPath(path string) (reporef.DatasetRef, error) {
+	refstr := HTTPPathToQriPath(path)
+	return repo.ParseDatasetRef(refstr)
+}
```
Comment (b5) [inline lib/api.go:142]: don't add imports to `repo/ref`, use `dsref.Parse` instead

File: api/context.go [hunk 2]
```diff
@@ -3,9 +3,8 @@ package api
 import (
 	"context"
 	"net/http"
-	"strings"
 
-	"github.com/qri-io/qri/repo"
+	"github.com/qri-io/qri/lib"
```
Comment (b5) [inline api/context.go:7]: well shoot, we should just move this entire file into lib in a future PR

File: api/serialization.go [hunk 3]
```diff
@@ -1,18 +1,9 @@
...
-	"path/filepath"
-	"strings"
 
 	"github.com/qri-io/dataset"
-	"github.com/qri-io/dataset/detect"
-	"github.com/qri-io/qri/base/fill"
-	"gopkg.in/yaml.v2"
 )
 
 // inlineScriptsToBytes consumes all open script files for dataset components
```
Comment (b5) [inline api/serialization.go:9]: very little left of this file after moving the `formFileDataset` func, let's move `inlineScriptToBytes` into `api/dataset.go`, the only place it's currently used

File: lib/api.go (hunk 1 above)
Comment (b5) [reply lib/api.go:142]: That said, I'm a big fan of moving these functions down into lib

**Review by b5 — APPROVED** [review]
Looking _great!_. I'm a fan of the few bits you've moved down into lib, and this `UnmarshalFromRequest` pattern seems to be showing it's value.

File: api/context.go (hunk 2 above)
Comment (Arqu) [reply api/context.go:7]: Lol, yes, just added a todo for that. Should happen rather soon.

**Pull request merged**